pkg/phlaredb/block_querier.go

```diff
@@ -1627,6 +1639,9 @@ func (b *singleBlockQuerier) SelectMergeByLabels(
 	if err := b.Open(ctx); err != nil {
 		return nil, err
 	}
+	b.queries.Add(1)
```

Comment:
I feel like we should update Open to be a function with a call back like `Query(fn ...) error` and all of this is handled there.